src/main/java/com/yourorg/TrackJavaTodos.java

```diff
@@ -1,50 +1,42 @@
+/*
+ * Copyright 2025 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
 package com.yourorg;
 
+import com.yourorg.table.TodoCommentsReport;
 import lombok.EqualsAndHashCode;
 import lombok.Value;
-import org.jspecify.annotations.Nullable;
-import org.openrewrite.*;
+import org.openrewrite.Recipe;
 
-import java.util.LinkedHashSet;
-
-// TODO - This is a placeholder for a scanning recipe.
-// Implement a recipe that finds any comments in Java source files that contain `TODO`, and add them to a file called `TODO.md`.
+// TODO - This is a placeholder for a recipe that uses data tables.
+// Implement a recipe that finds any comments in Java source files that contain `TODO`, and add them to a data table.
 // You're done when all of the tests in `TrackJavaTodosTest` pass.
 @Value
 @EqualsAndHashCode(callSuper = false)
-public class TrackJavaTodos extends ScanningRecipe<TrackJavaTodos.TodoComments> {
+public class TrackJavaTodos extends Recipe {
+
+    transient TodoCommentsReport todoCommentsTable = new TodoCommentsReport(this);
```

Comment:
I wonder if we're giving away too much by already including this reference. I've seen folks use `DataTable<TodoCommentsReport.Row>` here which is a helpful learning exercise.
```suggestion
    transient TodoCommentsReport todoCommentsTable = new TodoCommentsReport(this);
```